tests/adapters/test_protocol.py

```diff
@@ -22,10 +22,177 @@
 # THE SOFTWARE.
 #
 
-from pymeasure.adapters.protocol import ProtocolAdapter
+from pymeasure.adapters.protocol import to_bytes, ProtocolAdapter
+
+from pytest import mark, raises, fixture
+
+
+@mark.parametrize("input, output", (("superXY", b"superXY"),
+                                    ([1, 2, 3, 4], b"\x01\x02\x03\x04"),
+                                    (5, b"5"),
+                                    (4.6, b"4.6")
+                                    ))
+def test_to_bytes(input, output):
+    assert to_bytes(input) == output
+
+
+def test_to_bytes_invalid():
+    with raises(TypeError):
+        to_bytes(5.5j)
 
 
 def test_protocol_instantiation():
-    a = ProtocolAdapter()
-    assert not a
+    a = ProtocolAdapter([("write", "read"), ("write_only",)])
+    assert a.comm_pairs == [("write", "read"), ("write_only",)]
     # TODO: Make this first test assert something useful
+
+
+class Test_write:
+    @mark.parametrize("command, written", (("hey Ho", b"hey Ho"),
+                                           (b"superX", "superX")
+                                           ))
+    def test_write(self, command, written):
+        a = ProtocolAdapter([(written, 5)])
+        a.write(command)
+        assert a._read_buffer == b"5"
+
+    def test_write_without_response(self):
+        a = ProtocolAdapter([("Hey ho",)])
```

Comment:
As I said above, I think it will make everything more consistent if a one-tuple becomes disallowed, and we used `("Hey ho", None)` here.